### ISSUE TITLE
Use default value as example in OAS if no example is explicitly provided

### ISSUE THREAD

**Issue by JohnPaton** [tiangolo/fastapi#1176]
### Is your feature request related to a problem

Kind of. I've seen a number of issues like #1082, #723 where people don't know exactly how to specify examples for their OAS docs. I made #1106 to improve the docs a bit on this point, but it seems like it could be more intuitive at the API level as well.

### The solution you would like

My proposal is to select the `example` as follows:
* Example provided explicitly: Use that value
* Elif default value provided but no example: Use the default value as the example (or leave it out of the example if the default is `None`)
* Required field: example must be set explicitly

### Describe alternatives you've considered

Improving the docs could help, but I think the API for examples could be cleaner too which is a "deeper" issue.

### Additional context

A lot of my models currently look like

```python
class Item(BaseModel):
    name: str = Field('some-name', example='some-name')
    id: int = Field(0, example=0)
    # etc. etc.
```

This change would help model definitions to be more DRY.

I'd be happy to contribute a PR if maintainers are open to this change.

**Comment by tiangolo** [comment]
Thanks for the suggestion! But actually, the examples shown in the docs are generated automatically by Swagger UI.

FastAPI is not generating automatic examples for docs.

But what you could do is create a class method `schema_extra` as described in https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization

And that `schema_extra` would add your custom examples using the default values from the model.

And you could probably add it to your own `CustomBaseModel`, and then have all your models inherit from it.

**Comment by JohnPaton** [comment]
Hey! So there is always an example in the Swagger UI right? It's just that if the field is an `int` then the example value is `0`, if it's a `str` then the example is `"string"`, etc. What I'm suggesting is that if a field has a default value, and it doesn't have an explicit example set, then we take the default value as the example instead of `0`, `"string"`, etc. 

It would make the examples (which are quite an important part of the docs IMO) much more intuituve for no extra work from the user.

**Comment by mcauto** [comment]
It is a good idea to augment related documents.

but It's not fastapi feature.

you should suggest to pydantic.

**Comment by tiangolo** [comment]
Yeah, as @mcauto says, that actually comes from Pydantic directly.

But a main blocker right now is that OpenAPI, in the current version, uses `example`, that's what Swagger UI uses. But JSON Schema in modern versions, and OpenAPI 3.1.0 (not released yet), use `examples`. So adding a strong default like that would end up in backward-compatibility issues :disappointed: 

Nevertheless, you could do a more or less simple trick, create a custom Pydantic model that inherits from `BaseModel`, and add a function to handle schema customization: https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization

It would take care of generating the `example` in the schema automatically from the default values.

And then you could inherit from that for your models.

That would solve the use case in a simple way without incurring into near-future backward compatibility issues.
